### ISSUE TITLE
I'm having problems with z_endstop

### ISSUE THREAD

**Issue by Eugenio-Andreola** [ralf-e/ANET-3D-Board-V1.0#4]
Hello folks, I'm having problems with the endstop of the Z axis, it is always active, I saw that I would need to redo the soldering of the capacitor (R39), but if I change it, what is the most correct specification for it?

Ceramic capacitor Smd 10uf 50v 20% X7r 1.7mm X 0.8mm

Ceramic capacitor Smd 10uf 50v 10% X7r1210 3.2x2.5mm

Or would there be another one more correct?

Taking advantage of this, I'm thinking of replacing the C45 resistor as well.

Remembering that I know practically nothing about electronics and soldering ehehe

Thanks for the help.

**Comment by hannesweisbach** [comment]
Hey Eugenio,

R usually designates resistors, not capacitors. According to the schematic R39 has a value of 10k. Run-of-the mill resistors usually have a voltage rating of 50V, which is more than enough for this application. If you have any other resistors handy, I'd say anything from 4.7k to 47k is probably ok. The higher the resistance, the more suceptible to noise the endstop will be. 10k is good tradeoff between current consumption and noise susceptibility.

The size is not denoted in the schematic and I don't have the board in front of me. I guess it's 0805 or 0603.


Best,
Hannes

**Comment by Eugenio-Andreola** [comment]
Oops, I messed up with the information...
How are you, Hannesweisbach?
I was really referring to the capacitor (C45) and not the resistor (R39) which apparently is not showing any problems until now. But I'm thinking of changing the two, regarding the Capacitor, would you know how to help me and understand which would be the best?
In the specifications they have some data that I don't know what the percentage means: Ceramic capacitor Smd 10uf 50v 20% X7r
Ceramic capacitor Smd 10uf 50v 10% X7r, would it mean their accuracy?
And this X7r, would it be some information that I would need to pay attention to in order to choose the correct one?

**Comment by hannesweisbach** [comment]
Hey,

No problem. I had a look at the schematic again. In my opinion C45 is optional, unless you're having problems with your end stop, like "ghost triggering", when your printer says your end stop activated when it clearly didn't.
The capacitor is purely to remove noise from the signal line. IMHO 10u are *way* too much. I think 10nF are more suitable. The reason is the capacitor is charged to 5V through R39 and then the end stop switch shorts out C45. The (relatively) high and sudden current from a 10u ceramic capacitor will wear out and break the contacts in the end stop (especially cheap ones).
Think contacts or distributors in the ignition of older cars.
This is much less of a problem if you have solid state end stop, such as optical or magnetic (hall effect) ones.

The letters such as X7R and Y5V refer the electrolytic used in the cap. X7R is better than Y5V. NP0 is better still (and more expensive). Y5V is sufficient for this application, but since Y5V is the same price as X7R, especially in onesie quantities, it doesn't matter. (X7R used to be more expensive than Y5V). TLDR: X7R is ok.

The percentage means how much the capacitance can vary under nominal conditions (normal temperature, "normal" voltage (depends on the type, usually they are measured at 0.6V or something?). So let's say 100nF 20% can go from 80nF to 120nF (or was it 80nF to 100nF, could be the percentage is only how much less the capacitance could be). Capacitance also reduces with voltage and temperature. At nominal voltage it may very well be the capacitor has only 20-40% of it's nominal capacitance left.
TLDR: In this application it doesn't matter. Choose the cheapest.

I'd use 10-47nF, Y5V or X7R (or some unrated crap in my leftover box), 10V+ rating. 50V is probably overrated too. You can also try if it work without this cap. This depends on your steppers, their configuration, wire routing etc.
